src/compiler/scala/tools/nsc/backend/jvm/MethodNode1.java

```
@@ -0,0 +1,39 @@
+package scala.tools.nsc.backend.jvm;
+
+import scala.tools.asm.Label;
+import scala.tools.asm.Opcodes;
+import scala.tools.asm.tree.LabelNode;
+import scala.tools.asm.tree.MethodNode;
+
+public class MethodNode1 extends MethodNode {
+    public MethodNode1(int api, int access, String name, String descriptor, String signature, String[] exceptions) {
+        super(api, access, name, descriptor, signature, exceptions);
+    }
+
+    public MethodNode1(int access, String name, String descriptor, String signature, String[] exceptions) {
+        this(Opcodes.ASM6, access, name, descriptor, signature, exceptions);
+    }
+
+    public MethodNode1(int api) {
+        super(api);
+    }
+
+    public MethodNode1() {
+        this(Opcodes.ASM6);
+//        instructions = new InsnList() {
+//            @Override
+//            public void add(AbstractInsnNode insnNode) {
+//                // TODO assert(insnNode.index == -1)
+//                super.add(insnNode);
+//            }
+//        };
+    }
+
+    @Override
+    protected LabelNode getLabelNode(Label label) {
+        if (!(label.info instanceof LabelNode)) {
+            label.info = new LabelNode1(label);
+        }
+        return (LabelNode) label.info;
```

I tried pushing this into ASM itself, but doing so breaks a [unit test](https://github.com/retronym/asm/blob/2152b6abc5a49e6bdc2cfd00a23ceac064317c55/asm-tree/src/test/java/org/objectweb/asm/tree/ClassNodeTest.java#L360-L374) which appears to be a legitimate use case the only works under the status quo.

The failure mode for that test is an infinite loop iterating an instruction list. (or, a fail fast from our other customization that adds an assertion in `InsnList.add` to check that the given instruction is not already in another instruction list.